code-challenges.test.js

```diff
@@ -17,78 +17,173 @@
+    })
 
+    var mixedDataArray2 = [3, "yo!", 94, -9, false, 0, 18, "hola!"]
+    var mixedDataArray2Expected = [-9, 0, 3, 18, 94]
+    // Expected output: [-9, 0, 3, 18, 94]
+    it('return [-9, 0, 3, 18, 94] for an input of [3, "yo!", 94, -9, false, 0, 18, "hola!"]', () => {
+        expect(onlyNumbersSorted(mixedDataArray2)).toEqual(mixedDataArray2Expected);
+    })
+})
 
 
 // b) Create the function that makes the test pass.
 
-
+// Create a function that takes in 1 argument that is an array
+const onlyNumbersSorted = (array) => {
+    // Use .filter() to go through the array and check which ones are numbers
+    // Store that in a variable called 'numArray'
+    numArray = array.filter(value => {
+        return typeof value === 'number';
+    })
+
+    // Sort the number array in ascending order
+    // using an arrow function and the ternary operator
+    // to check whether elements a and b need to be swapped.
+    // After the ternary operator '?', the first expression
+    // is the truthy, and the expression after the colon ':'
+    // is the falsey.
+    // For .sort, if the compareFunction returns a value less
+    // than zero, then leave a and b unchanged.
+    // If the compareFunction returns a value greater than 0,
+    // sort b before a.
+    return numArray.sort( (a, b) => a > b ? 1 : -1)
+}
 
 
 
 // --------------------4) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-var vowelTester1 = "learn"
-// Expected output: 1
-var vowelTester2 = "academy"
-// Expected output: 0
-var vowelTester3 = "challenge"
-// Expected output: 2
-
-
-
+// Name of the function will be firstVowel
+describe("firstVowel", () => {
+    var vowelTester1 = "learn"
+    // Expected output: 1
+    it("return the number 1 for an input of string 'learn'", () => {
+        expect(firstVowel(vowelTester1)).toEqual(1);
+    })
+    var vowelTester2 = "academy"
+    // Expected output: 0
+    it("return the number 0 for an input of string 'academy'", () => {
+        expect(firstVowel(vowelTester2)).toEqual(0);
+    })
+    var vowelTester3 = "challenge"
+    // Expected output: 2
+    it("return the number 2 for an input of string 'challenge'", () => {
+        expect(firstVowel(vowelTester3)).toEqual(2);
+    })
+    var vowelTester4 = "bcdfghjklmnpqrstvwxyz"
+    // Expected output: No vowels were found
+    it("return the string 'No vowels were found' for the string 'bcdfghjklmnpqrstvwxyz'", () => {
+        expect(firstVowel(vowelTester4)).toEqual('No vowels were found');
+    })
+})
 
 
 // b) Create the function that makes the test pass.
+
+// Create a function that takes in 1 argument that is a string
+const firstVowel = (string) => {
+    // Split the string into an array
+    let strArray = string.split('');
+    // Iterate through the string to look for the first vowel
+    for (let i = 0; i < strArray.length; i++) {
```

Comment:
great optimization. Using a for-loop will allow you only iterate for only as long as you need to run the loop!